apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -1554,11 +1629,124 @@ where
+                                                    return Ok(SelectionMapperReturn::None);
+                                                }
+                                            }
+                                        }
+                                        Ok(SelectionMapperReturn::Selection(selection.clone()))
+                                    },
+                                )?;
+                                // TODO: What happens if we can't convert `e` into an EdgeIndex?
+                                let resolution = condition_resolver.resolve(
+                                    (*e).into().unwrap(),
+                                    context,
+                                    excluded_destinations,
+                                    excluded_conditions,
+                                )?;
+                                // TODO: Is this still needed?
+                                // assert(edge.transition.kind === 'FieldCollection', () => `Expected edge to be a FieldCollection edge, got ${edge.transition.kind}`);
+                                let Some(arg_indices) =
+                                    self.graph.subgraph_to_arg_indices.get(&ctx.subgraph_name)
+                                else {
+                                    internal_error!("TODO")
+                                };
+                                let Some(id) = arg_indices.get(&ctx.argument_coordinate).cloned()
+                                else {
+                                    internal_error!("TODO")
+                                };
+
+                                // TODO: This logic was in the JS code. Does a -1 cost me the
+                                // resolution was unsatisfied?
+                                // if (resolution.cost === -1 || totalCost === -1) {
+                                //  totalCost = -1;
+                                match &resolution {
+                                    ConditionResolution::Satisfied {
+                                        cost, path_tree, ..
+                                    } => {
+                                        total_cost += cost;
+                                        let entry = ContextMapEntry {
+                                            levels_in_data_path,
+                                            levels_in_query_path,
+                                            path_tree: path_tree.clone(),
+                                            selection_set,
+                                            inbound_edge: (*e).into().unwrap(),
+                                            param_name: ctx.named_parameter.clone(),
+                                            arg_type: ctx.arg_type.clone(),
+                                            id,
+                                        };
+                                        // TODO: Do we need to check for map collisions here?
```

Comment:
The JS code didn't check for this, but I want to double check. Is `named_parameter` unique?
```suggestion
```

Reply:
If that's the case, we should be using the `ContextCondition`'s `argument_coordinate` not the `named_parameter`. The named parameter is just a string.

Reply:
The named parameter represents an argument name, and in this function, we're (among other things) resolving conditions imposed by the `@fromContext`s on the arguments in the field edge handed to the function. This means the named parameters here are from the same field and unique, so they won't collide.